### DISCUSSION TITLE
Suddenly unable to connect to Tshock hosted Terraria world

### DISCUSSION THREAD

**Discussion by rpeters004** [Pryaxis/TShock#2570] [Q&A]
So I initially followed some tutorial for getting a Tshock Terraria server up and running including port forwarding and whatnot, and it worked pretty much first try and has worked well for at least several weeks now. I leave it running on a spare computer so my brother and I can join from separate locations at any time, and do co-op stuff. 

Today I was at work, and texted my brother to check out something I had done on the world. He did and it worked. Later on, he tried to get on again and it wouldn't work. I get home, try to join as well, same problem. Stuck on connecting to xxxx:xxx..... no dice. The server is still running on my computer as it has been for weeks, and I wasn't around to change anything. No changes to firewall settings, etc. Since then I've tried restarting my computers, exiting and running the server again, restarting my router etc. 

I'm confused how it suddenly stopped working. I'm not terribly tech savvy, so I don't know if any updates to Terraria or Tshock on their end could have changed anything. I've heard that ISPs can sometimes block ports. Is that a possible explanation. If I check on canyouseeme.org I get connection refused on 7777, and timed out on any other port. Let me know if any additional information would be helpful. Thanks!

**Comment by Arthri** [comment]
Is Terraria allowed through your firewall?

**Reply by rpeters004** [reply]
So I never had to make any changes to the firewall initially to make it work. I didn't make any changes today either, between when my brother was able to get on and then was not able to get on. Just to be sure, I turned windows defender off completely, and still no luck :(

**Reply by Arthri** [reply]
What do you mean by "I turned windows defender off"? Do you mean this? 
![image](https://user-images.githubusercontent.com/41360489/155060750-741089cd-7b56-410b-8afa-9395b588dbbc.png)

Windows Defender and Firewall and very separate things and disabling either is very risky. I suggest you look up how to add a port through firewall rather than disabling it completely

**Reply by rpeters004** [reply]
I just temporarily turned off the Windows Defender Firewall to eliminate that as a possible source of the problem, and the problem of being unable to connect persisted after turning it off. It's back on now.

**Reply by rpeters004** [reply]
Per your suggestion, I have now added port 7777 to be allowed through the firewall, although that was never necessary before. Still not working, unfortunately.

**Reply by Arthri** [reply]
Oh. Yeah I'm not sure then. I suspect it's something in between your connection that secretly changed(like you mentioned, your ISP blocking ports).

**Reply by rpeters004** [reply]
Yeah I'm afraid of that. What are the odds that anyone I can get on the phone with at Spectrum will have any clue what I'm talking about...

**Reply by Arthri** [reply]
I neither live in America nor have a lot of experience with ISPs, so I don't really know, sorry.

**Reply by rpeters004** [reply]
No problem, thanks for your help.
